### PR TITLE
Goja JSVM for Gateway

### PR DESCRIPTION
**Pull request by joshblakeley**
For https://github.com/TykTechnologies/tyk/issues/1893

This PR introduces a new JSVM that can be used in Tyk. 

In gateway config value is `enable_v2_jsvm`

### CODE REVIEW

**Comment by joshblakeley** [comment]
@matiasinsaurralde would appreciate you looking over the plugin side of things - test all pass locally but are a bit flaky on ci still

**Comment by matiasinsaurralde** [comment]
Looks like a vendoring issue to me, related to `golang.org/x/text/language`.

**Comment by joshblakeley** [comment]
Yep i just broke it now when vendoring - it was a different issue before related to a timeout or hang. I'll fix vendoring then ping you again 👍

**Comment by matiasinsaurralde** [comment]
We should also check the latest `otto` changes, I've updated the vendored version in my local branch and seems there's some [new stuff](https://github.com/robertkrimen/otto/commits/master), mostly bugfixes:
```
diff --git a/vendor/vendor.json b/vendor/vendor.json
index 9f5b84a5..d8bdd32c 100644
--- a/vendor/vendor.json
+++ b/vendor/vendor.json
@@ -437,10 +437,10 @@
                {
-                       "checksumSHA1": "ZSYig6eYkc34ELHl6O5gRtJvr6o=",
+                       "checksumSHA1": "WdorbyEJiNbjIOiQzM4TLb/XA7Q=",
                        "path": "github.com/robertkrimen/otto",
-                       "revision": "fc2eb1bbf1c5c462313c810bb02dc93cd964aebe",
-                       "revisionTime": "2017-07-21T19:43:36Z"
+                       "revision": "15f95af6e78dcd2030d8195a138bd88d4f403546",
+                       "revisionTime": "2018-06-17T13:11:54Z"
                },
```

**Comment by joshblakeley** [comment]
@matiasinsaurralde PTAL

File: main.go [hunk 1]
```diff
@@ -466,13 +466,17 @@ func loadCustomMiddleware(spec *APISpec) ([]string, apidef.MiddlewareDefinition,
 	mwPreFuncs := []apidef.MiddlewareDefinition{}
 	mwPostFuncs := []apidef.MiddlewareDefinition{}
 	mwPostKeyAuthFuncs := []apidef.MiddlewareDefinition{}
-	mwDriver := apidef.OttoDriver
-
+	var mwDriver apidef.MiddlewareDriver
```
Comment (matiasinsaurralde) [inline main.go]: We could default to `otto` here and simplify the `if` block:
```
mwDriver := apidef.OttoDriver
if config.Global().JSVM == "goja" {
  mwDriver = apidef.GojaDriver
}
```

**Comment by buger** [comment]
I wonder if we can refactor it a bit, since jsvm_goja.go and jsvm_otto.go contains almost idential code.

**Comment by joshblakeley** [comment]
I'll see where I can squeeze it down

**Pull request closed without merging**